tests/conftest.py

```diff
@@ -0,0 +1,8 @@
+import pytest
+from unittest.mock import patch
+
+
+@pytest.fixture(scope='session')
```

Comment:
can this function be moved to `test_clustering.py`? I think it is no need to be a 'session' scope function.